src/main/java/application/bookstore/repository/book/BookSpecificationProviderManager.java

```diff
@@ -0,0 +1,26 @@
+package application.bookstore.repository.book;
+
+import application.bookstore.model.Book;
+import application.bookstore.repository.SpecificationProvider;
+import application.bookstore.repository.SpecificationProviderManager;
+import java.util.List;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BookSpecificationProviderManager implements SpecificationProviderManager<Book> {
+    private List<SpecificationProvider<Book>> bookSpecificationProviders;
+
+    public BookSpecificationProviderManager(List<SpecificationProvider<Book>>
+                                                    bookSpecificationProviders) {
+        this.bookSpecificationProviders = bookSpecificationProviders;
+    }
+
+    @Override
+    public SpecificationProvider<Book> getSpecificationProvider(String key) {
+        return bookSpecificationProviders.stream()
+                .filter(p -> p.getKey().equals(key))
```

Comment:
avoid short one letter naming even in stream chains, readability is more important that a second you save not writing out that full name